### PR TITLE
Cap # of NFTs minted (and not burned)

### LINKED ISSUE #1812: Add property to cap # of NFTs minted in current price regime
### Problem

Only a fixed number of NFTs will be allowed to be minted in the current pricing regime. 

### Solution

Add a global/dynamic  property `tokens.nfts.maxAllowedMints=5000000` and resolve to `MAX_NFTS_IN_PRICE_REGIME_HAVE_BEEN_MINTED` for any `TokenMint` that attempts to increase the number of NFTs in state beyond this limit.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
- Closes #1812

**Notes for reviewer**:
- Uses a new global/dynamic `tokens.nfts.maxAllowedMints=5000000` property to (effectively) limit the size of the `uniqueTokens` map.
- A `TokenMint` with a non-empty `metadata` list will fail with `MAX_NFTS_IN_PRICE_REGIME_HAVE_BEEN_MINTED` if the `uniqueTokens.size() + metadataList.size() > tokens.nfts.maxAllowedMint`.
- Add `TokenManagementSpecs.nftMintingCapIsEnforced()` EET to validate dynamic changes to `tokens.nfts.maxAllowedMint`.

_Other minor improvements_
- Delete unused `BackingNfts` from the `TypedTokenStore`.
- Update `MerkleUniqueTokenId.hashCode()` to not use `Objects.hash()`, which takes varargs and is inappropriate on a performance-critical path.

**Checklist**
- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by qnswirlds — APPROVED** [review]
LGTM

File: hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java [hunk 1]
```diff
@@ -100,6 +105,13 @@ public void doStateTransition() {
 		accountStore.persistAccount(token.getTreasury());
 	}
 
+	private void validateMinting(Token token, TokenMintTransactionBody op) {
+		if (token.getType() == NON_FUNGIBLE_UNIQUE) {
+			final var proposedTotal = tokenStore.currentMintedNfts() + op.getMetadataCount();
+			validateTrue(validator.isPermissibleTotalNfts(proposedTotal), MAX_NFTS_IN_PRICE_REGIME_HAVE_BEEN_MINTED);
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java:111]: nit: based on the context, `MAX_NFTS_IN_PRICE_REGIME_HAVE_BEEN_MINTED` is a little vague here. Maybe `MORE_THAN_MAX_NFTS_IN_PRICE_REGIME_HAVE_BEEN_MINTED` or something like that will be more explicit to tell this is an error condition.

**Review by ljianghedera — APPROVED** [review]
LGTM

File: hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java (hunk 1 above)
Comment (qnswirlds) [reply hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java:111]: No, it means max has been reached and we cannot mint any more. Saying more than max nfts have been minted is actually incorrect.

Comment (ljianghedera) [reply hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java:111]: Got it. My understanding was wrong. Thx for the clarification.

**Review by anighanta — APPROVED** [review]

**Review by abhishek-hedera — APPROVED** [review]
LGTM

**Pull request merged**